### PR TITLE
[5.1] Make refreshing model on create an optional trait.

### PR DESCRIPTION
**Pull request by crynobone**
This make refreshing model after creating an option, just like "soft deletes".

Related to #10392

Signed-off-by: crynobone crynobone@gmail.com

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Much better idea. :+1:

File: src/Illuminate/Database/Eloquent/RefreshOnCreate.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        static::created(function ($model) {
+            static::refresh($model);
+        });
+    }
+
+    /**
+     * Refresh the current model with the current attributes from the database.
+     *
+     * @param  \Illuminate\Database\Eloquent\Model  $model
+     *
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/RefreshOnCreate.php]: cs

**Comment by crynobone** [comment]
CS fixed (and squashed).

**Comment by arrilot** [comment]
In my opinion this doesn't actually solve anything.  It's still confusing because its not on by default.
You still need to google about this trait and you can simply manually refresh the model at that moment

**Comment by arrilot** [comment]
Another problem is that if you enable this trait, it starts hitting you everywhere. 

To sum up I would rather have it turned on by default and have an ability to disable it on runtime

**Comment by GrahamCampbell** [comment]
I think it should definitely not be enabled by default.

**Comment by lukasgeiter** [comment]
@arrilot Consider this: Many developers are never going to realize that the framework makes an additional (in many cases unnecessary) query and won't disable it. On the other hand, if we make it opt-in, people will notice that something with DB defaults doesn't work and will google / search the docs.

**Comment by crynobone** [comment]
Why isn't soft deletes enabled by default? Is it confusing to enable soft
deletes if you needs it?

**Comment by darthmaim** [comment]
:+1: This is the better solution, many developers don't need to reload the models and can save a query to the database. Just add one sentence to the docs in the creating models section about this trait and it should be good.

**Comment by taylorotwell** [comment]
I think we should just stick with specifying default attributes in the `$attributes` property.

**Pull request closed without merging**